src/core/thread/mle_router.cpp

```diff
@@ -4311,6 +4316,9 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Header *aHeader, Message *aMe
         mRouters[GetLeaderId()].SetNextHop(GetRouterId(mParent.GetRloc16()));
     }
 
+    // send announce after become Router if needed
+    InformPreviousChannel();
```

Comment:
Move this to below `exit:` in case anything in handling the address response fails?

Reply:
Updated, Thanks.